// See NextBeatClientAsync for docstrings/comments

syntax = "proto3";

package nextbeat;

message AddJobsRequest {
    bytes pickled_job_definitions = 1;
}

message AddJobsResponse {
    string status = 1;
}

message EventsRequest {
    repeated string topic_names = 1;
}

message Events {
    bytes pickled_events = 1;
}

message RegisterJobRunnerRequest {
    string job_runner_type = 1;
    string address = 2;
}

message RegisterJobRunnerResponse {
}

message ManualRunRequest {
    string job_name = 1;
}

message ManualRunResponse {
}

service NextBeatServer {
    rpc add_jobs(AddJobsRequest) returns (AddJobsResponse) {};

    rpc get_events(EventsRequest) returns (Events) {};

    rpc register_job_runner(RegisterJobRunnerRequest) returns
        (RegisterJobRunnerResponse) {};

    rpc manual_run(ManualRunRequest) returns (ManualRunResponse) {};
}
